### PR TITLE
Remove unnecessary else clauses in for loops

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

File: sphinx/environment/__init__.py [hunk 1]
```diff
@@ -354,9 +354,7 @@ def path2doc(self, filename):
         for suffix in self.config.source_suffix:
             if fnmatch.fnmatch(filename, '*' + suffix):
                 return filename[:-len(suffix)]
-        else:
-            # the file does not have docname
```
Comment (tk0miya) [inline sphinx/environment/__init__.py:358]: please don't remove this comment.

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM with ntis

File: sphinx/environment/__init__.py (hunk 1 above)
Comment (srinivasreddy) [reply sphinx/environment/__init__.py:358]: Done. Kept it as is.

**Review by tk0miya — APPROVED** [review]
Thanks for quick work!

**Comment by tk0miya** [comment]
Merged! :-)
Thank you for contribution!

**Pull request merged**